operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -349,20 +349,29 @@ private String createPatchString() {
 
     private Step createUpdateSteps() {
       final Step next = createDomainStatusReplaceStep();
-      EventData eventData = createDomainEvent();
-      return eventData == null ? next : Step.chain(EventHelper.createEventStep(eventData), next);
+      ArrayList<EventData> eventDataList = createDomainEvents();
+      return eventDataList.isEmpty() ? next : Step.chain(createEventSteps(eventDataList), next);
     }
 
-    @Nullable
-    EventData createDomainEvent() {
+    private Step createEventSteps(ArrayList<EventData> eventDataList) {
+      Step nextStep = null;
+      for (int i = eventDataList.size() - 1; i >= 0; i--) {
```

Comment:
With the current approach, you have two different places in the code that need to be coordinated in an odd way to get the right ordering. A much simpler approach would be to use the natural ordering of the created events and then define a sort order for them, so that no matter the order in which you add them to the list, iterating them will produce the desired order.

Reply:
I like the suggestion. I'll give it a try.